### PR TITLE
Feat json schema

### PR DESCRIPTION
**Pull request by roytev**
## what
* Added a json schema for the charts values
* Added missing customizations to the values.yaml
* Added missing customizations to the Readme.md table

## why
* make sure that if we change something in the values yaml we are not breaking something.
* when user install or upgrade it gets more verbose information if he put something wrong in the values.
* also when user make typo the additionalproperties:false will let him know that the attribute is not supported.
* when using intelliJ and i presume other IDE's its much easier to make changes as we are seeing the description from the schema.
* its a preparation of an effort of mine to make this chart more robust and maintainable. i also want to add helm unit tests 
*  
## tests
* ran helm lint with all possible values.yaml attribute 

## references
* https://www.arthurkoziel.com/validate-helm-chart-values-with-json-schemas/
* https://austindewey.com/2020/06/13/helm-tricks-input-validation-with-values-schema-json/
* https://helm.sh/docs/topics/charts/#schema-files

### CODE REVIEW

**Comment by jamengual** [comment]
could you bump the version please?

**Comment by roytev** [comment]
@jamengual Thanks done!

**Review by GMartinez-Sisti — COMMENTED** [review]
Thanks for the PR @roytev! Left some comments/suggestions.

File: charts/atlantis/values.yaml [hunk 1]
```diff
@@ -216,10 +229,11 @@ readinessProbe:
 
 service:
   type: NodePort
+  annotations: {}
   port: 80
-  nodePort: null
+  nodePort: ~
```
[the remark this answers is not in the document]
Comment (roytev) [reply charts/atlantis/values.yaml]: Agreed replaced all occurrences

**Comment by roytev** [comment]
@GMartinez-Sisti Thanks! ive made the requested changes

File: charts/atlantis/values.schema.json [hunk 2]
```diff
@@ -0,0 +1,2360 @@
...
+          "description":"Allow customizing securityContext.",
+          "$ref":"#/definitions/io.k8s.api.core.v1.PodSecurityContext"
+        },
+        "priorityClassName":{
+          "description":"Leverage a PriorityClass to ensure your pods survive resource shortages.",
+          "type":[
+            "string",
+            "null"
+          ]
+        },
```
Comment (GMartinez-Sisti) [inline charts/atlantis/values.schema.json:544]: I think this also needs to change?
```suggestion
        "priorityClassName":{
          "description":"Leverage a PriorityClass to ensure your pods survive resource shortages.",
          "type": "string"
        },
```

Comment (roytev) [reply charts/atlantis/values.schema.json:544]: yes to be consistent i agree

**Review by GMartinez-Sisti — APPROVED** [review]
LGTM 😄

**Review by jamengual — DISMISSED** [review]

**Review by jamengual — APPROVED** [review]

**Pull request merged**